File: adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/importer/ImporterParameterDescription.java
package org.jvalue.ods.adapterservice.adapter.importer;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@EqualsAndHashCode
@AllArgsConstructor
public class ImporterParameterDescription {
  private String name;
  private String description;
  private boolean required;
  private Class<?> type;

  public ImporterParameterDescription(String name, String description, Class<?> type) {
    this(name, description, true, type);
  }
}

File: adapter/src/adapter/importer/ImporterParameterDescription.ts
export class ImporterParameterDescription {
  name: string;
  description: string;
  required: boolean;
  type: unknown;

  constructor({
    name,
    description,
    required = true,
    type,
  }: {
    name: string;
    description: string;
    required?: boolean;
    type: unknown;
  }) {
    this.name = name;
    this.description = description;
    this.required = required;
    this.type = type;
  }
}
